src/goto-symex/symex_start_thread.cpp

```diff
@@ -68,10 +68,9 @@ void goto_symext::symex_start_thread(statet &state)
     lhs.set_level_0(t);
 
     // set up L1 name
-    if(!state.level1.current_names.insert(
-        std::make_pair(lhs.get_l1_object_identifier(),
-                       std::make_pair(lhs, 0))).second)
-      UNREACHABLE;
+    auto emplace_result = state.level1.current_names.emplace(
+      lhs.get_l1_object_identifier(), std::make_pair(lhs, 0));
```

Comment:
Until we have C++17 use:
````c++
auto emplace_result = state.level1.current_names.emplace(
  std::piecewise_construct,
  std::forward_as_tuple(lhs.get_l1_object_identifier()),
  std::forward_as_tuple(lhs, 0));
````
to avoid even constructing the second pair if the key already exists.